services/order.py

```diff
@@ -0,0 +1,46 @@
+from datetime import datetime
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, User
+
+
+def create_order(
+    tickets: list[dict],
+    username: str,
+    date: datetime = None
+) -> Order:
+    with transaction.atomic():
+        user = User.objects.get(username=username)
+        order = Order.objects.create(
+            user=user,
+        )
+
+        if date:
+            order.created_at = date
+
+        order.save()
+
+        for ticket_data in tickets:
+            row = ticket_data["row"]
+            seat = ticket_data["seat"]
+            movie_session_id = ticket_data["movie_session"]
+
+            Ticket.objects.create(
+                order=order,
+                row=row,
+                seat=seat,
+                movie_session_id=movie_session_id
+            )
+
+    return order
```

Comment:
When you apply transaction.atomic() to the whole function body, it is a better choice to use it as a decorator instead of a context manager.